app/models/user.rb

```diff
@@ -4,13 +4,37 @@ class User < ActiveRecord::Base
     config.authentications_class = Authentication
   end
 
-  has_many :cards
+  has_many :decks
+
+  belongs_to :current_deck, class_name: "Deck"
+  
+  validate :current_deck_belongs_to_decks, on: :set_current_deck
+
+  def current_deck_belongs_to_deck
+    if decks.include?(deck.find(current_deck_id))
+      errors.add(:current_deck, "Deck id and current deck id cant be different" ) 
```

Comment:
Line is too long. [82/80]<br>Space inside parentheses detected.<br>Trailing whitespace detected.